src/DIRAC/RequestManagementSystem/Client/ReqClient.py

```diff
@@ -688,3 +691,52 @@ def recoverableRequest(request):
                         return False
                     return True
     return True
+
+
+class _JobDBInteraction:
+    """Class to handle the JobDB interaction.
+
+    This will either connect to the DB directly or use the client depending on
+    if use_certificates is set or not.
+
+    WARNING: This is not intended for use outside of ReqClient!
+    """
+
+    def __init__(self, useCertificates: bool):
+        self._useCertificates = useCertificates
+
+    def setJobParameter(self, jobID: int, key: str, value: str):
+        if self._useCertificates:
+            vo = returnValueOrRaise(self._jobStatusUtility.jobDB.getJobAttribute(jobID, "VO"))
+            return self._elasticJobParametersDB.setJobParameter(int(jobID), key, value, vo=vo)
+        else:
+            return self._client.setJobParameter(jobID, key, value)
```

Comment:
To draw attention to this, I think `JobStateUpdateClient().setJobParameter` no longer works with a host certificate as the VO isn't knonw.

cc @fstagni